common/lib/dependabot/dependency_group.rb

```diff
@@ -9,12 +9,21 @@
 
 module Dependabot
   class DependencyGroup
-    attr_reader :name, :rules, :dependencies
+    attr_reader :name, :rules, :dependencies, :handled_dependencies
 
     def initialize(name:, rules:)
       @name = name
       @rules = rules
       @dependencies = []
+      @handled_dependencies = Set.new
+    end
+
+    def add_to_handled(*dependencies)
+      @handled_dependencies += dependencies.map(&:name)
```

Comment:
🤔 is it sufficient to store just the `name`? Some package managers, like npm, allow having multiple versions of the same dependency installed.

In that case, if they're defined in the `package.json`, I do you need to name them explicitly, and that distinct name is likely what we get here, for multiple versions in the `package-lock.json` (transitive dependencies), this likely ends up not being an issue?

Reply:
Hmmm good question. Is it currently possible for Dependabot to handle (create, update) multiple PRs for the same dependency name? I would think not, at least based on how the [job dependencies](https://github.com/dependabot/dependabot-core/blob/23de1c7583117bd15f26ab33482383ed57208f14/updater/lib/dependabot/job.rb#L89) property for rebasing contains only the name and not some kind of unique identifier.

Reply:
Yep, I think you're right 👍